### DISCUSSION TITLE
Mirth service stopping after getting JIT debugger error

### DISCUSSION THREAD

**Discussion by adi-09** [nextgenhealthcare/connect#5646] [Q&A] [answered]
Whenever we are trying to launch the mirth admin 3.4.2, it is throwing Just In Time Debugger Exception, and when clicked on ok the service stops. Registry keys cannot be edited since it is a client machine.
![Screenshot 2023-01-31 140029](https://user-images.githubusercontent.com/55851132/215708446-e1a1eda2-d57c-4337-ab82-87cdb0287075.png)
System specifications -
OS: windows server 12
mirth version : 3.4.2

Any inputs to resolve this would be helpful. Thanks in advance.

**Comment by ab-20-m** [comment, marked as answer]
You appear to be using a non-standard/highly customized implementation of Mirth. [System.BadImageFormatException](https://learn.microsoft.com/en-us/dotnet/api/system.badimageformatexception?view=net-7.0#remarks) is a .NET exception that gets thrown when a DLL/EXE getting loaded doesn't match what the runtime expects (32-bit vs 64-bit). Mirth Connect is a Java application and doesn't use .NET. 

The releases of Mirth already come with a service wrapper. If I had to guess someone took a release of [Windows Service Wrapper](https://github.com/winsw/winsw) and decided to use that instead. Evidence for that scenario would be a `mcservice.xml` or `mcservice.exe.config` file being located beside the `mcservice.exe` file.
